app/api/utils/appContextMiddleware.ts

```diff
@@ -4,6 +4,7 @@ import { appContext } from 'api/utils/AppContext';
 const appContextMiddleware = (_req: Request, _res: Response, next: NextFunction) => {
   appContext
     .run(async () => {
+      appContext.set('requestId', Math.floor(Math.random() * 10000));
```

Comment:
@mfacar i think this is better placed in its own middleware like the multitenant one ?

Reply:
added requestIdMiddleware